src/main/java/com/sap/oss/phosphor/fosstars/data/DataProvider.java

```diff
@@ -65,6 +65,8 @@
    */
   Set<Feature<?>> supportedFeatures();
 
+  boolean supports(Object type);
```

Comment:
Let's please try to avoid using `Object`.

```suggestion
  boolean supports(T type);
```

Or, maybe it is even better if we get rid of `T` and use `Subject` instead of it:

```java
public interface DataProvider {
  /**
   * Gathers data about an object and updates a specified {@link ValueSet}.
   *
   * @param object The object.
   * @param values The set of values to be updated.
   * @return This data provider.
   * @throws IOException If something went wrong.
   */
  DataProvider update(Subject subject, ValueSet values) throws IOException;
...

  boolean supports(Subject subject);
...
```

Reply:
> Let's please try to avoid using `Object`.

Completely agree and I like the idea using the `Subject`.
Have updated the PR/PoC accordingly.